src/vs/editor/browser/editorBrowser.ts

```diff
@@ -251,6 +251,47 @@ export interface IOverlayWidget {
 	getPosition(): IOverlayWidgetPosition | null;
 }
 
+/**
+ * A glyph margin widget renders in the editor glyph margin.
+ */
+export interface IGlyphMarginWidget {
+	/**
+	 * The class name to apply to the glyph widget.
+	 */
+	className: string;
```

Comment:
What is the purpose of `className` given that API users can pass in their own dom node where they can already define its `className` as they wish?